pkg/client/testing/core/fixture.go

```diff
@@ -49,153 +58,158 @@ type ObjectScheme interface {
 	runtime.ObjectTyper
 }
 
-// ObjectReaction returns a ReactionFunc that takes a generic action string of the form
-// <verb>-<resource> or <verb>-<subresource>-<resource> and attempts to return a runtime
-// Object or error that matches the requested action. For instance, list-replicationControllers
-// should attempt to return a list of replication controllers. This method delegates to the
-// ObjectRetriever interface to satisfy retrieval of lists or retrieval of single items.
-// TODO: add support for sub resources
-func ObjectReaction(o ObjectRetriever, mapper meta.RESTMapper) ReactionFunc {
+// ObjectReaction returns a ReactionFunc that applies core.Action to
+// the given registry.
+func ObjectReaction(registry Registry, mapper meta.RESTMapper) ReactionFunc {
 	return func(action Action) (bool, runtime.Object, error) {
+		ns := action.GetNamespace()
 		resource := action.GetResource()
+
 		kind, err := mapper.KindFor(resource)
+		if err != nil {
+			return false, nil, fmt.Errorf("unrecognized action for resource %s: %s", resource, err)
```

Comment:
"no kind for registered for resource"?  I know its pre-existing, but you've already tweaked it some.